File: solutions/fortran/raindrops/1/raindrops.f90
module raindrops
  implicit none
contains

  function convert(i)
    integer :: i, last
    character(20) :: convert
    convert=''
    last=1
    if (mod(i,3)==0) then
      convert=convert(1:last)//'Pling'
      last=last+5
    end if
    if (mod(i,5)==0) then
      convert=convert(1:last)//'Plang'
      last=last+5
    end if
    if (mod(i,7)==0) then
      convert=convert(1:last)//'Plong'
      last=last+5
    end if
    if (last==1) then
      write(convert, '(I10)') i 
    end if
    convert=adjustl(convert)
  end function convert

end module raindrops

File: solutions/cpp/raindrops/1/raindrops.cpp
#include "raindrops.h"

namespace raindrops {
    std::string convert(int n){
        std::string answer="";
        if (n<3)
            answer=std::to_string(n);
        else if (n%3 == 0 || n%5 == 0 || n%7 == 0){
            if (n%3 == 0)
                answer.append("Pling");
            if (n%5 == 0)
                answer.append("Plang");
            if (n%7 == 0)
                answer.append("Plong");  
        }
        else
            answer=std::to_string(n);
        return answer;
    }
}  // namespace raindrops
